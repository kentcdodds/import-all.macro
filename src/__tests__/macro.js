import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'
import prettier from 'prettier'
import {prettier as prettierConfig} from 'kcd-scripts/config'

const projectRoot = path.join(__dirname, '../../').replace(/\\/g, '/')

expect.addSnapshotSerializer({
  print(val) {
    return val
      .split(projectRoot)
      .join('<PROJECT_ROOT>/')
      .replace(/fixtures/g, 'files')
      .replace(/..\/macro/, 'import-all.macro')
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, prettierConfig)
  },
  tests: {
    'no usage': `import importAll from '../macro'`,
    'incorrect API usage': {
      error: true,
      code: `
        import importAll from '../macro'
        const x = importAll.defered('hi')
      `,
    },
    'non-static evaluate-able expression': {
      error: true,
      code: `
        import importAll from '../macro'
        const x = importAll(global.whatever)
      `,
    },
    'README:1 `importAll` uses dynamic import': `
      import importAll from '../macro'

      document.getElementById('load-stuff').addEventListener('click', () => {
        importAll('./fixtures/*.js').then(all => {
          console.log(all)
        })
      })
    `,
    'README:2 `importAll.sync` uses static imports': `
      import importAll from '../macro'

      const a = importAll.sync('./fixtures/*.js')
    `,
    'README:3 `importAll.deferred` gives an object with dynamic imports': `
      import importAll from '../macro'

      const routes = importAll.deferred('./fixtures/*.js')
    `,
  },
})
