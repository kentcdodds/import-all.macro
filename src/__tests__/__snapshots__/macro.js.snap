// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros README:1 \`importAll\` uses dynamic import: README:1 \`importAll\` uses dynamic import 1`] = `

import importAll from 'import-all.macro'

document.getElementById('load-stuff').addEventListener('click', () => {
  importAll('./files/*.js').then(all => {
    console.log(all)
  })
})

      ↓ ↓ ↓ ↓ ↓ ↓

document.getElementById('load-stuff').addEventListener('click', () => {
  Promise.all([
    import('./files/a.js'),
    import('./files/b.js'),
    import('./files/c.js'),
    import('./files/d.js'),
  ])
    .then(function importAllHandler(importVals) {
      return {
        './files/a.js': importVals[0],
        './files/b.js': importVals[1],
        './files/c.js': importVals[2],
        './files/d.js': importVals[3],
      }
    })
    .then(all => {
      console.log(all)
    })
})


`;

exports[`macros README:2 \`importAll.sync\` uses static imports: README:2 \`importAll.sync\` uses static imports 1`] = `

import importAll from 'import-all.macro'

const a = importAll.sync('./files/*.js')

      ↓ ↓ ↓ ↓ ↓ ↓

import * as _filesAJs from './files/a.js'
import * as _filesBJs from './files/b.js'
import * as _filesCJs from './files/c.js'
import * as _filesDJs from './files/d.js'
const a = {
  './files/a.js': _filesAJs,
  './files/b.js': _filesBJs,
  './files/c.js': _filesCJs,
  './files/d.js': _filesDJs,
}


`;

exports[`macros README:3 \`importAll.deferred\` gives an object with dynamic imports: README:3 \`importAll.deferred\` gives an object with dynamic imports 1`] = `

import importAll from 'import-all.macro'

const routes = importAll.deferred('./files/*.js')

      ↓ ↓ ↓ ↓ ↓ ↓

const routes = {
  './files/a.js': function () {
    return import('./files/a.js')
  },
  './files/b.js': function () {
    return import('./files/b.js')
  },
  './files/c.js': function () {
    return import('./files/c.js')
  },
  './files/d.js': function () {
    return import('./files/d.js')
  },
}


`;

exports[`macros incorrect API usage: incorrect API usage 1`] = `

import importAll from 'import-all.macro'
const x = importAll.defered('hi')

      ↓ ↓ ↓ ↓ ↓ ↓

Error: <PROJECT_ROOT>/src/__tests__/macro.js: ../macro: This is not supported: \`importAll.defered\`. Please see the import-all.macro documentation

`;

exports[`macros no usage: no usage 1`] = `

import importAll from 'import-all.macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros non-static evaluate-able expression: non-static evaluate-able expression 1`] = `

import importAll from 'import-all.macro'
const x = importAll(global.whatever)

      ↓ ↓ ↓ ↓ ↓ ↓

Error: <PROJECT_ROOT>/src/__tests__/macro.js: ../macro: There was a problem evaluating the value of the argument for the code: importAll(global.whatever). If the value is dynamic, please make sure that its value is statically deterministic.

`;
